BDCSCli/Recipe.hs

```diff
@@ -141,6 +141,26 @@ writeCommit repo branch filename message content = do
     -- Create a new commit: repositoryCreateCommit
     Git.repositoryCreateCommit repo ref sig sig encoding message tree [parent_commit]
 
+-- | Read a commit and return a ByteString of the content
+-- TODO Return the commit message too
+readCommit :: Git.Repository -> T.Text -> T.Text -> Maybe T.Text -> IO BS.ByteString
+readCommit repo branch filename Nothing = do
+    let spec = T.pack $ printf "%s:%s" branch filename
```

Comment:
It's up to you which way is more readable.  Sometimes I like to get rid of these calls to pack with code like this:
```let spec = T.concat [branch, ":", filename]```